### ISSUE TITLE
ln: failed to create symbolic link '/var/www/cgi-bin': File exists

### ISSUE THREAD

**Issue by oliverkurth** [dperson/smokeping#24]
Hi!

I am getting this error when building. I already tried to work around it by using "ln -fs ..."  but then I am getting another error: "ln: /var/www/cgi-bin: cannot overwrite directory".

...
Running hooks in /etc/ca-certificates/update.d...
done.
Enabling cgi: ok
Run "service lighttpd force-reload" to enable changes
Enabling fastcgi: ok
Run "service lighttpd force-reload" to enable changes
ln: failed to create symbolic link '/var/www/cgi-bin': File exists
The command '/bin/sh -c export DEBIAN_FRONTEND='noninteractive' &&     apt-get update -qq &&     apt-get install -qqy --no-install-recommends ca-certificates curl dnsutils                 echoping fonts-dejavu-core lighttpd procps smokeping ssmtp                 $(apt-get -s dist-upgrade|awk '/^Inst.*ecurity/ {print $2}') &&    apt-get clean &&     /bin/echo -e '+ EchoPingHttp\n\nbinary = /usr/bin/echoping\n'                 >>/etc/smokeping/config.d/Probes &&     /bin/echo -e '+ EchoPingHttps\n\nbinary = /usr/bin/echoping\n'                 >>/etc/smokeping/config.d/Probes &&     sed -i '/^syslogfacility/s/^/#/' /etc/smokeping/config.d/General &&     conf=/etc/lighttpd/lighttpd.conf dir=/etc/lighttpd/conf-available                 header=setenv.add-response-header &&     sed -i '/server.errorlog/s|^|#|' $conf &&     sed -i '/server.document-root/s|/html||' $conf &&     sed -i '/mod_rewrite/a\ \t"mod_setenv",' $conf &&     echo "\\n$header"' += ( "X-XSS-Protection" => "1; mode=block" )' >>$conf &&    echo "$header"' += ( "X-Content-Type-Options" => "nosniff" )' >>$conf &&     echo "$header"' += ( "X-Robots-Tag" => "none" )' >>$conf&&     echo "$header"' += ( "X-Frame-Options" => "SAMEORIGIN" )' >>$conf &&     echo '\n$HTTP["url"] =~ "^/smokeping($|/)" {' >>$conf &&     echo '\tdir-listing.activate = "disable"\n}' >>$conf &&     echo '\n# redirect to the right Smokeping URI' >>$conf &&     echo 'url.redirect  = ("^/$" => "/smokeping/smokeping.cgi",' >>$conf &&     echo '\t\t\t"^/smokeping/?$" => "/smokeping/smokeping.cgi")' >>$conf &&     sed -i 's|var/log/lighttpd/access.log|tmp/log|' $dir/10-accesslog.conf &&     sed -i '/^#cgi\.assign/,$s/^#//; /"\.pl"/i\ \t".cgi"  => "/usr/bin/perl",'                 $dir/10-cgi.conf &&     echo '\nfastcgi.server += ( ".cgi" =>\n\t((' >>$dir/10-fastcgi.conf &&     sed -i -e '/CHILDREN/s/[0-9][0-9]*/16/'                 -e '/max-procs/a\ \t\t"idle-timeout" => 20,'                 $dir/15-fastcgi-php.conf &&     grep -q 'allow-x-send-file' $dir/15-fastcgi-php.conf || {         sed -i '/idle-timeout/a\ \t\t"allow-x-send-file" => "enable",'                     $dir/15-fastcgi-php.conf &&         sed -i '/"bin-environment"/a\ \t\t\t"MOD_X_SENDFILE2_ENABLED" => "1",'                     $dir/15-fastcgi-php.conf; } &&     echo '\t\t"socket" => "/tmp/perl.socket" + var.PID,'                 >>$dir/10-fastcgi.conf &&     echo '\t\t"bin-path" => "/usr/lib/cgi-bin/smokeping.cgi",'                >>$dir/10-fastcgi.conf &&     echo '\t\t"docroot" => "/var/www",' >>$dir/10-fastcgi.conf &&     echo '\t\t"check-local"     => "disable",\n\t))\n)'                 >>$dir/10-fastcgi.conf &&     unset conf dir header &&     lighttpd-enable-mod cgi &&     lighttpd-enable-mod fastcgi &&     [ -d /var/cache/smokeping ] || mkdir -p /var/cache/smokeping &&     [ -d /var/lib/smokeping ] || mkdir -p /var/lib/smokeping &&     [ -d /run/smokeping ] || mkdir -p /run/smokeping &&     ln -s /usr/share/smokeping/www /var/www/smokeping &&     ln -s /usr/lib/cgi-bin /var/www/ &&     ln -s /usr/lib/cgi-bin/smokeping.cgi /var/www/smokeping/ &&     chown -Rh smokeping:www-data /var/cache/smokeping /var/lib/smokeping                 /run/smokeping &&     chmod -R g+ws /var/cache/smokeping /var/lib/smokeping /run/smokeping &&     chmod u+s /usr/bin/fping &&     rm -rf /var/lib/apt/lists/* /tmp/*' returned a non-zero code: 1

**Comment by dperson** [comment]
Have you changed anything? It's building fine for me and the official docker build servers. Here's the [last build log](https://hub.docker.com/r/dperson/smokeping/builds/bk2dxudhemgatx3cjfrbqfh/).

**Comment by oliverkurth** [comment]
I haven't changed anything when I hit the error. I removed the link command, and then it built without errors. However, I don't see any activity in the graphs, even when setting SPUSER=root (since I have seen that issue before, this may or may not be related).

I looked at the build.log at https://hub.docker.com/r/dperson/smokeping/builds/bk2dxudhemgatx3cjfrbqfh/ and the package versions for lighttpd are different - I have this:

Selecting previously unselected package lighttpd.
Preparing to unpack .../113-lighttpd_1.4.45-1_amd64.deb ...

Other packages may be different too, since the Dockerfile uses current debian:stretch, and the log at hub.docker.com is 15 days old. I attached the build log, generated after reverting my change. git status is clean, and it's at 840d319070bf8f658a141f39b30e21c5f86451cc .

[build.txt](https://github.com/dperson/smokeping/files/743334/build.txt)
